components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -754,6 +754,7 @@ export abstract class AbstractTypeORMWorkspaceDBImpl implements WorkspaceDB {
         query = query.orderBy("pws.creationTime", "DESC");
         query = query.innerJoinAndMapOne("pws.workspace", DBWorkspace, "ws", "pws.buildWorkspaceId = ws.id");
         query = query.andWhere("ws.deleted = false");
+        query = query.andWhere("ws.contentDeletedTime = ''");
```

Comment:
Will try this in prod (offline) to get a feeling for cardinality...

Reply:
Currently trying this in prod-failover with the official GitLab repo (with and without the fix). Will add findings to the PR description.

Reply:
Reduced cardinality from 16919 to 337. :muscle: